sw/host/opentitanlib/src/spiflash/flash.rs

```diff
@@ -208,7 +208,19 @@ impl SpiFlash {
     }
 
     /// Read into `buffer` from the SPI flash starting at `address`.
-    pub fn read(&self, spi: &dyn Target, mut address: u32, buffer: &mut [u8]) -> Result<()> {
+    pub fn read(&self, spi: &dyn Target, address: u32, buffer: &mut [u8]) -> Result<()> {
+        self.read_with_progress(spi, address, buffer, |_| {})
+    }
+
+    /// Read into `buffer` from the SPI flash starting at `address`.
+    /// The `progress` callback will be invoked after each chunk of the read operation.
+    pub fn read_with_progress<F: FnMut(u32)>(
+        &self,
+        spi: &dyn Target,
+        mut address: u32,
+        buffer: &mut [u8],
+        mut progress: F,
```

Comment:
Mumble I would use `impl Fn(u32)` but up to you.

Reply:
Done.  (actually Fn(u32, u32), taking address and chunksize to make the progress updating functions easier).